moby/moby
# Remove local fork of archive/tar package

**- What I did**
Removed the local fork of `archive/tar` while retaining all existing mitigations of CVE-2019-14271. Fixes #42402

**- How I did it**
I hid the value containing the file's UID and GID from the `tar.FileInfoHeader` function [so that it would elide calling `user.LookupId`](https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/archive/tar/stat_unix.go;l=28-31), analogous to the proposal in golang/go#50102.

**- How to verify it**
1. Comment out [the other mitigation](https://github.com/moby/moby/blob/559ff0ac661c1ca27bb17765629f6166d90ff499/pkg/chrootarchive/archive.go#L15-L20) and [the mitigation introduced in this PR](https://github.com/corhere/moby/blob/0ef67c6a9bdc7420210318fcef4fe0af3fc0cac6/pkg/archive/archive.go#L413-L421) and `make dynbinary`
2. Build a container image which exploits CVE-2019-14271 (an exploit payload which prints a message to stderr and exits nonzero makes it easy to spot in `dockerd`'s debug logs) and create a container from that image
3. `docker cp` from the container and observe that the exploit code has been executed
4. Uncomment the mitigation from this PR (leaving the other mitigation commented out), rebuild dockerd, and try to exploit the vulnerability again. Observe that the exploit payload does not execute.

**- Description for the changelog**
Remove local fork of `archive/tar` package

**- A picture of a cute animal (not mandatory but encouraged)**
